File: smartreminderssave/src/main/java/jgappsandgames/smartreminderssave/tasks/Checkpoint.java
package jgappsandgames.smartreminderssave.tasks;

// JSON
import org.json.JSONException;
import org.json.JSONObject;

/**
 * Checkpoint
 * Created by joshua on 10/4/17.
 */
public class Checkpoint {
    private static final String ID = "position";
    private static final String STATUS = "status";
    private static final String TEXT = "text";

    public int id;
    public boolean status;
    public String text;

    @SuppressWarnings("unused")
    public Checkpoint(int id) {
        this.id = id;
        status = false;
    }

    public Checkpoint(int id, @SuppressWarnings("SameParameterValue") String text) {
        this.id = id;
        this.text = text;
        status = false;
    }

    public Checkpoint(JSONObject object) {
        if (object == null) return;

        id = object.optInt(ID, 0);
        status = object.optBoolean(STATUS, false);
        text = object.optString(TEXT, "");
    }

    public JSONObject toJSON() {
        try {
            JSONObject data = new JSONObject();
            data.put(ID, id);
            data.put(STATUS, status);
            data.put(TEXT, text);

            return data;
        } catch (JSONException j) {
            j.printStackTrace();
            return null;
        }
    }

    @Override
    public String toString() {
        return toJSON().toString();
    }
}

File: smartreminderssave/src/main/java/jgappsandgames/smartreminderssave/tasks/Checkpoint.kt
package jgappsandgames.smartreminderssave.tasks

// JSON
import org.json.JSONException
import org.json.JSONObject

/**
 * Checkpoint
 * Created by joshua on 12/12/2017.
 */
// TODO: Move into the Task File
class Checkpoint(var id: Int, var text: String, var status: Boolean) {
    companion object {
        // Constants -------------------------------------------------------------------------------
        private const val ID = "position"
        private const val STATUS = "status"
        private const val TEXT = "text"
    }

    // Constructors --------------------------------------------------------------------------------
    constructor(c_id: Int, c_text: String): this(c_id, c_text, false)
    constructor(data: JSONObject): this(data.optInt(ID, 0), data.optString(TEXT, ""), data.optBoolean(STATUS, false))

    // Json Methods --------------------------------------------------------------------------------
    fun toJSON(): JSONObject? {
        return try {
            val data = JSONObject()
            data.put(ID, id)
            data.put(STATUS, status)
            data.put(TEXT, text)

            data
        } catch (j: JSONException) {
            j.printStackTrace()
            null
        }

    }

    // String Method -------------------------------------------------------------------------------
    override fun toString(): String {
        return toJSON()!!.toString()
    }
}